dali/operators/segmentation/biased_crop_center.cc

```diff
@@ -0,0 +1,180 @@
+
+  std::vector<int> foreground_;
+  TensorListShape<> crop_shape_;
+
+  bool has_crop_shape_ = false;
+
+  USE_OPERATOR_MEMBERS();
+};
+
+BiasedCropCenterCPU::BiasedCropCenterCPU(const OpSpec &spec)
+    : Operator<CPUBackend>(spec), rng_(spec.GetArgument<int64_t>("seed")) {
+}
+
+bool BiasedCropCenterCPU::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                    const workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  output_desc.resize(1);
+  output_desc[0].shape = uniform_list_shape(nsamples, {ndim});
+  output_desc[0].type = TypeTable::GetTypeInfo(DALI_INT64);
+
+  foreground_.clear();
+  if (spec_.HasTensorArgument("foreground")) {
+    foreground_.resize(nsamples);
+    const auto &foreground_arg_in = ws.ArgumentInput("foreground");
+    for (int i = 0; i < nsamples; i++) {
+      foreground_[i] = *foreground_arg_in[i].data<int>();
+    }
+  } else {
+    foreground_.resize(nsamples, spec_.GetArgument<int>("foreground"));
+  }
+
+  has_crop_shape_ = spec_.ArgumentDefined("shape");
+  if (has_crop_shape_) {
+    GetShapeArgument(crop_shape_, spec_, "shape", ws, ndim, nsamples);
+  }
+  return true;
+}
+
+template <typename T>
+void BiasedCropCenterCPU::RunImplTyped(workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  auto &out_center = ws.template OutputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  auto masks_view = view<const T>(in_masks);
+  auto center_view = view<int64_t>(out_center);
+  auto& thread_pool = ws.GetThreadPool();
+
+  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+    thread_pool.AddWork(
+      [&, sample_idx](int thread_id) {
+        auto mask = masks_view[sample_idx];
+        auto center = center_view[sample_idx];
+
+        const auto &mask_sh = mask.shape;
+        auto crop_sh = crop_shape_.tensor_shape_span(sample_idx);
+
+        if (foreground_[sample_idx]) {
+          SearchableRLEMask rle_mask(mask);
+          auto dist = std::uniform_int_distribution<int64_t>(0, rle_mask.count() - 1);
+          int64_t flat_idx = rle_mask.find(dist(rng_));
+
+          // Convert from flat_idx to per-dim indices
+          auto mask_strides = kernels::GetStrides(mask_sh);
+          for (int d = 0; d < ndim - 1; d++) {
+            center.data[d] = flat_idx / mask_strides[d];
+            flat_idx = flat_idx % mask_strides[d];
+          }
+          center.data[ndim - 1] = flat_idx;
+
+          // Adjust center if necessary
+          if (has_crop_shape_) {
+            for (int d = 0; d < ndim; d++) {
+              int64_t w = crop_sh[d] >> 1;
+              center.data[d] =
+                  boundary::idx_clamp(center.data[d], w, mask_sh[d] - (crop_sh[d] - w));
```

Comment:
```suggestion
                  clamp(center.data[d], w, mask_sh[d] - (crop_sh[d] - w));
```
If you're just clamping, you don't need any fancy boundary handling (ordinary clamp will do).
Also, boundary clamp is an _exclusive_ clamp (to size-1) and we want to have inclusive clamp.
Example:
mask_sh[d] = 100
crop_sh[d] = 10
w = 5
maximum valid value is 95 (the window will start at 90 and end at 99)